consolidate game-playing bots

fredfishgames opened the pull request:
This is for issue #283 
I've rewritten the game adapter so that it supports multiple games at once.
People can make games in streams, or invite players through PMS, where the bot creates a subject for a game.
All the unit tests are run by the `game_handler_test_bot`, but I could merge them into the connect four tests if that is better.
Thanks

Edit: I've added mypy annotations for `game_handler.py`

To do (GCI Task):

- [x] Create game adapter to support multiple games
- [x] Implement new game adapter to connect_four
- [x] Write tests for game adapter
- [x] Write tests for connect four

To do (in general):
- [ ] Create test library
- [x] Migrate tictactoe
- [ ] Write tests for tictactoe
- [ ] Migrate chess
- [ ] Migrate merels
- [ ] Create documentation

eeshangarg commented on zulip_bots/zulip_bots/bots/connect_four/connect_four.py:
```diff
@@ -51,6 +52,8 @@ def __init__(self):
 
-        super(ConnectFourBotHandler, self).__init__(game_name, bot_name, move_help_message, move_regex, model, gameMessageHandler)
+        super(ConnectFourBotHandler, self).__init__(game_name, bot_name,
+                                                    move_help_message, move_regex, model, gameMessageHandler, max_players=2)
```
Could you please wrap the arguments at 80 characters?

eeshangarg commented on zulip_bots/zulip_bots/bots/connect_four/test_connect_four.py:
```diff
@@ -4,18 +4,20 @@
 
     # Function that serves similar purpose to BotTestCase.verify_dialog, but allows for multiple responses to be handled
-    def verify_response(self, request, expected_response, response_number, data=None, computer_move=None, user = 'foo@example.com'):
+    def verify_response(self, request, expected_response, response_number, data=None, user='foo@example.com'):
```
Again, try to wrap the arguments at 80 characters.

And add mypy annotations for all of this code please. Otherwise the build will fail :)

eeshangarg commented on zulip_bots/zulip_bots/bots/game_handler_test_bot/game_handler_test_bot.py:
```diff
@@ -0,0 +1,59 @@
+        gameMessageHandler = GameHandlerTestBotMessageHandler
+
+        super(GameHandlerTestBotHandler, self).__init__(game_name, bot_name,
+                                                        move_help_message, move_regex, model, gameMessageHandler, max_players=2, supports_computer=True)
```
Again, please wrap the arguments at 80 chars. :)

fredfishgames commented:
Thanks, I've changed the names of the commits, added mypy, and fixed line sizes. Once it is ready to be merged I'll rewrite all the commits to get rid of game_handler improvements.

showell commented:
Merged, thanks @fredfishgames!
